make_test_data.py

```diff
@@ -239,14 +245,64 @@ def get_taxonomy(self, num_orfs: int = 2):
             "names": names,
         }
 
+    def get_bed_alignments(self, num_contigs=1):
+        contig_col = 0
+        coverages = pd.read_csv(
+            self.bed_fpath, sep="\t", index_col=contig_col, header=None
+        )
+        # Get number of unique contigs set by `num_contigs`, default is 1.
+        contigs = set(coverages.index.unique().tolist()[:num_contigs])
+        coverages = coverages.loc[contigs]
+        coverages.reset_index(
+            inplace=True
+        )  # Here we are ready to send to json object self.data
+        return coverages
+
+    def get_sam_alignments(self, num_contigs=1):
+        with open(self.sam_fpath) as fh:
+            lines = ""
+            contig = None
+            for line in fh:
+                if line.startswith("@SQ"):
+                    # example line
+                    # @SQ	SN:NODE_1503_length_7231_cov_222.076	LN:7231
+                    contig = line.split("\t")[1]
+                    contig = contig.replace("SN:", "")
+                if line.startswith("@HD") or line.startswith("@PG") or contig in line:
+                    # @HD and @PG is required for bam construction
+                    lines += line
+        return lines
+
+    def get_reads(self, read_count=5):
+        reads = []
+        for file in [self.fwd_reads, self.rev_read]:
+            outlines = ""
+            count = 0
+            with open(file) as fh:
+                for line in fh:
+                    if "+" in line:
+                        count += 1
+                    if count >= read_count:
+                        break
+                    outlines += line
+            reads.append(outlines)
+        return reads
+
     def get_coverage(self):
-        if "coverage" not in self.data:
-            self.data["coverage"] = {
-                "spades_records": self.metagenome,
-                "bam": "alignments.bam",
-                "sam": "alignments.sam",
-            }
-        return self.data["coverage"]
+        logging.info("Making alignment records (sam, and bed files) ...")
+        bed = self.get_bed_alignments()
+        sam = self.get_sam_alignments()
+        logging.info("Getting fwd and rev reads ...")
+        reads = self.get_reads()
+        fwd_reads = reads[0]
+        rev_reads = reads[1]
```

Comment:
Let's do tuple unpacking here to keep retrieval of any piece of test data to one line

```suggestion
        fwd_reads, rev_reads = self.get_reads()
```